src/components/popup/HeadV2.tsx

```diff
@@ -101,12 +109,33 @@ export default function HeadV2({
   // history back
   const [, goBack] = useHistory();
 
+  // Display an application logo if `url` is provided:
+
+  const [appInfo, setAppInfo] = useState<AppInfo>({});
+
+  useEffect(() => {
+    async function loadAppInfo() {
+      if (!url) return;
+
+      const app = new Application(url);
+      const appInfo = await app.getAppData();
+
+      setAppInfo(appInfo);
+    }
+
+    loadAppInfo();
+  }, [url]);
+
+  const appName = appInfo?.name || url;
+  const appIconPlaceholderText = appName?.slice(0, 2).toUpperCase();
```

Comment:
If `url` is used as a fallback, we should get the first 2 letters from the domain, not from the subdomain.

Reply:
True. This can also be added as a separate issue

Reply:
https://github.com/arconnectio/ArConnect/issues/570